avocado/plugins/runner_nrunner.py

```diff
@@ -151,9 +152,44 @@ def _get_all_runtime_tasks(test_suite):
                              runnable.uri,
                              None,
                              no_digits)
+            # handles the test task
             task = nrunner.Task(runnable, test_id, status_uris,
                                 nrunner.RUNNERS_REGISTRY_PYTHON_CLASS)
-            runtime_tasks.append(RuntimeTask(task))
+            runtime_task = RuntimeTask(task)
```

Comment:
Consider creating a `_get_requirement_tasks()` of sorts here to improve the readability and maintainability.